### PR TITLE
Implement porter uninstall

### LINKED ISSUE #243: porter uninstall
`porter uninstall [CLAIM] [--insecure] [-f bundle.json] [--cred azure] [--param header=blue] [--param-file dev-values]`

All the arguments and flags work the same as `porter install` where everything can be defaulted when in the directory of the bundle.

### PR DESCRIPTION
**Pull request by carolynvs**
Closed #243

### CODE REVIEW

File: build/testdata/bundles/wordpress/porter.yaml [hunk 1]
```diff
@@ -42,5 +42,6 @@ install:
 uninstall:
 - helm:
     description: "Uninstall Wordpress"
-    name:
-      source: bundle.parameters.wordpress-name
+    purge: true
+    releases:
+      - "source: bundle.parameters.wordpress-name"
```
Comment (vdice) [inline build/testdata/bundles/wordpress/porter.yaml]: Ooh, this works?  Handy to know...

File: pkg/porter/cnab.go [hunk 2]
```diff
@@ -1,10 +1,262 @@
...
 type CNABProvider interface {
 	Install(arguments cnabprovider.InstallArguments) error
-	//Upgrade() error
-	//Uninstall() error
+	Uninstall(arguments cnabprovider.UninstallArguments) error
+}
+
+
+// sharedOptions are common options that apply to multiple CNAB actions.
+type sharedOptions struct {
```
Comment (vdice) [inline pkg/porter/cnab.go]: sharing is caring!

File: build/testdata/bundles/wordpress/porter.yaml (hunk 1 above)
Comment (carolynvs) [reply build/testdata/bundles/wordpress/porter.yaml]: When you are in the yaml list, you need to put quotes around `"source: bundle...."` to make it work. But yeah it's neat!

File: pkg/porter/cnab.go (hunk 2 above)
Comment (carolynvs) [reply pkg/porter/cnab.go]: Yeah I think we'll want to do this for the duffle Install/Uninstall args too but I wasn't confident so I held off on that for now. It seems that pretty much all the cnab actions take the same options and just treat things slightly differently. I didn't want to consolidate too quickly so I wanted to wait until we had all the actions implemented fully, then see if they are all the same.

**Pull request merged**